[ADD] account_exchange_difference

cav-adhoc opened the pull request.

cav-adhoc committed:
[ADD] account_exchange_difference_invoice: add module

roboadhoc commented:
[![Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/account-financial-tools/pull/776.png)](https://mergebot.adhoc.ar/ingadhoc/account-financial-tools/pull/776)

jjscarafia commented on account_exchange_difference_entry/models/account_move_line.py:
```diff
@@ -0,0 +1,57 @@
+            ("move_id.reversal_move_ids", "=", False),
+            ("move_id.reversed_entry_id", "=", False),
+            "&",
+            ("invoice_origin", "=ilike", "Exch reversal of%"),
```
podremos hacerlo de otra manera?

jjscarafia commented on account_exchange_difference_entry/models/account_payment.py:
```diff
@@ -0,0 +1,34 @@
+                note.line_ids._all_reconciled_lines().filtered(
+                    lambda l: l.matched_debit_ids or l.matched_credit_ids
+                ).filtered(lambda x: x.move_type == "entry").move_id.write(
+                    {"invoice_origin": f"Exch reversal of {note.name}"}
```
idem, me hace un poco de ruido lo del invoice_origin + campo "debit_credit_note_ids", lo pensaría mas genérico desde reversal del entry...

jjscarafia committed:
[IMP] _difference_invoice: Avoid two entries when unreconciling

Cambiamos approach para que si se descncilia un asiento con diff de cambio y no estaba convertido a factura, borramos asiento para:
1. evitar dominio complejo
2. evitar verbosidad de asientos (en menus, partner ledger y demás)

cav-adhoc committed:
[IMP] account_exchange_difference_invoice: add ref to original exchange
and simplify assing of the payment with _reconcile_reversed_moves

cav-adhoc committed:
[IMP] account_exchange_difference_invoice: avoid note if balance is zero

cav-adhoc commented on account_exchange_difference_invoice/models/account_move_line.py:
```diff
@@ -0,0 +1,109 @@
+                else:
+                    rec.exchange_info += " (Debit Note Pending)"
+
+            if rec.move_id in rec.move_id.exchange_reversed_move_ids:
```
Ver en mejora del campo exchange info

cav-adhoc committed:
[IMP] account_exchange_difference_invoice: copilot suggestions

jjscarafia commented:
@roboadhoc r+ rebase-ff

roboadhoc commented:
Merge method set to rebase and fast-forward.
